src/hyphenClient.ts

```diff
@@ -14,66 +14,56 @@ export class HyphenClient {
     this.cache = new CacheClient(options.cache);
   }
 
+  private async tryUrls(urlPath: string, payload: unknown, logger?: Logger): Promise<Response> {
+    let lastError: unknown;
+
+    for (const baseUrl of this.horizonServerUrls) {
+      try {
+        const url = `${baseUrl}${urlPath}`;
```

Comment:
you should probably https://developer.mozilla.org/en-US/docs/Web/API/URL/URL rather than string concatination.